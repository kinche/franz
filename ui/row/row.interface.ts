export interface RowProps {
  children: any
  style?: React.CSSProperties
}
