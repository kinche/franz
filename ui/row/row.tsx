import React, { PureComponent } from 'react'

// Interface
import { RowProps } from './row.interface'

export class Row extends PureComponent<RowProps> {
  render() {
    const { children, style } = this.props

    return (
      <div style={style}>
        {children}

        <style jsx={true}>{`
          div {
            max-width: 1200px;
            margin-left: auto;
            margin-right: auto;
            padding-left: 32px;
            padding-right: 32px;
          }
        `}</style>
      </div>
    )
  }
}
