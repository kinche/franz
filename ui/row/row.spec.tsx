/* eslint-env jest */

// Packages
import renderer from 'react-test-renderer'

// Components
import { Row } from './row'

// Interface
import { RowProps } from './row.interface'

const Component = (props: RowProps) => <Row {...props}>{props.children}</Row>

describe('Row', () => {
  it('should render row', () => {
    const component = renderer.create(<Component style={{ margin: 'auto' }}>children</Component>)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
