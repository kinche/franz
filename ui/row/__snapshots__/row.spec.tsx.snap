// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row should render row 1`] = `
<div
  className="jsx-2953685289"
  style={
    Object {
      "margin": "auto",
    }
  }
>
  children
</div>
`;
